Fix unexpected empty hash errors in popups (#4793)
AzureAD/microsoft-authentication-library-for-js: lib/msal-browser/docs/errors.md

1. You are being throttled by your identity provider
1. Your identity provider did not redirect back to your `redirectUri`.

> Note: If you are using a router, please make sure it does not strip the hash or auto-redirect while MSAL token acquisition is in progress. When possible it's best if your `redirectUri` page does not invoke the router at all.

Reviewer: ```suggestion
**Important**: If your application uses a router library (e.g. React Router, Angular Router), please make sure it does not strip the hash or auto-redirect while MSAL token acquisition is in progress. If possible, it is best if your `redirectUri` page does not invoke the router at all.
```

};
```

### hash_empty_error

**Error Messages**:

> - Hash value cannot be processed because it is empty. Please verify that your redirectUri is not clearing the hash.

Reviewer: ```suggestion
> Hash value cannot be processed because it is empty. Please verify that your redirectUri is not clearing the hash.
```

This error occurs when the page you use as your redirectUri is removing the hash, or auto-redirecting to another page. This most commonly happens when the application implements a router which navigates to another route, dropping the hash.

To resolve this error we recommend using a dedicated redirectUri page which is not subject to the router. For silent and popup calls it's best to use a blank page. If this is not possible please make sure the router does not navigate while MSAL token acquisition is in progress. You can do this by detecting if your application is loaded in an iframe for silent calls, in a popup for popup calls or by awaiting `handleRedirectPromise` for redirect calls.

### hash_does_not_contain_known_properties

**Error Messages**:

> - Hash does not contain known properites. Please verify that your redirectUri is not changing the hash.

Reviewer: ```suggestion
> Hash does not contain known properites. Please verify that your redirectUri is not changing the hash.
```
